packages/core/src/selectors/llmSelectors.ts

```diff
@@ -0,0 +1,162 @@
+// from typing import Any, Dict, List, Optional, Sequence, cast
+
+import { DefaultPromptTemplate } from "../extractors/prompts";
+import { LLM } from "../llm";
+import { SelectionOutputParser } from "../outputParsers/selectors";
+import { BaseOutputParser, QueryBundle, ToolMetadata } from "../types";
+import { BaseSelector, SelectorResult } from "./base";
+import { getDefaultSingleSelectPrompt } from "./prompts";
+
+function _buildChoicesText(choices: ToolMetadata[]): string {
+  const texts: string[] = [];
+  for (const [ind, choice] of choices.entries()) {
+    let text = choice.description.split("\n").join(" ");
+    text = `(${ind + 1}) ${text}`; // to one indexing
+    texts.push(text);
+  }
+  return texts.join("");
+}
+
+function _structuredOutputToSelectorResult(output: any): SelectorResult {
+  const structuredOutput = output;
+  const answers = structuredOutput.parsedOutput;
+
+  // adjust for zero indexing
+  const selections = answers.map((answer: any) => {
+    return { index: answer.choice - 1, reason: answer.reason };
+  });
+
+  return { selections };
+}
+
+type LLMPredictorType = LLM;
+
+/**
+ * A selector that uses the LLM to select a single or multiple choices from a list of choices.
+ */
+export class LLMMultiSelector extends BaseSelector {
```

Comment:
isn't that the same class as `LLMSingleSelector` with a different prompt?

Reply:
Yes, not sure yet what's the plan to expand this later, but if not we would change to an multi arg parameter